File: OOP/Classes & Objects/array_stack.java
import java.util.*;

class Stack{

	int[] arr;
	int len, top;

	void make_stack(int n){
		len = n;
		arr = new int[len];
		top = -1;
	}

	boolean isFull(){
		if(top + 1 >= len){
			return true;
		}
		else{
			return false;
		}
	}

	boolean isEmpty(){
		if(top == -1){
			return true;
		}
		else{
			return false;
		}
	}

	void push(int ele){
		if(isFull()){
			System.out.println("Stack Overflow!\n");
		}
		else{
			arr[++top]=ele;
			System.out.println(ele + "added to stack at" + top);
		}
	}

	void pop(){
		if(isEmpty()){
			System.out.println("Stack Undeflow!\n");
		}
		else{
			System.out.println(arr[top--] + "removed from" + (top+1));
		}
	}

	void peek(){
		if(top==-1){
			System.out.println("No element at the top!\n");
		}
		else{
			System.out.println(arr[top] + "is at the top\n");
		}
	}
}

class array_stack{
	public static void main(String[] args){

		Stack stk = new Stack();
		System.out.println("Enter length of stack\n");
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		stk.make_stack(n);
		char c; int choice;

		do{
			System.out.println("Choose action to perform\n1.Push\n2.Pop\n2.Peek\n");
			choice = sc.nextInt();

			switch(choice){
				case 1: System.out.println("Enter element to enter\n");
					int ele = sc.nextInt();
					stk.push(ele);
					break;

				case 2: stk.pop(); break; 

				case 3: stk.peek(); break;

				default: System.out.println("choose valid option\n"); break;
			}

			System.out.println("Continue? Yes or No\n");
			c = sc.next().charAt(0);
		} while(c=='Y' || c=='y');
	}
}

File: DS/Stacks/array_stack.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_LEN 100
#define true 1
#define false 0

typedef struct{
	char arr[MAX_LEN];
	int top;
} stack;

int isFull(stack* s){
	if(s->top==MAX_LEN-1) return true;
	else return false;
}

int isEmpty(stack *s){
	if(s->top==-1) return true;
	else return false;
}

void push(stack* s, char ele){
	if(!isFull(s)) {s->arr[++(s->top)] = ele; printf("%c pushed to stack at %d\n", ele, s->top);}
	else printf("Stack Overflow\n");
}

void pop(stack* s){
	if(!isEmpty(s)) {char ele = s->arr[(s->top)--]; printf("%c popped from stack at %d\n", ele, (s->top)+1);}
	else printf("Stack Underflow\n");
}

void display(stack* s){
	if(s->top==-1){
		printf("Stack is empty.\n");
	}
	int i=0;
	while(i<=s->top){
		printf("%c ", s->arr[i++]);
	}
	printf("\n");
}

int main(void){
	stack* s;
	s = (stack*)malloc(sizeof(stack));
	s->top = -1;
	char ch;	
	do{

		printf("Choose action to perform\n1.Push\n2.Pop\n3.Display\n");
		int choice; scanf("%d", &choice); 
		switch(choice){
			case 1: 
				printf("Enter element to push\n");
				char e; 
				scanf(" %c", &e);
				push(s,e);
				break;

			case 2:
				pop(s); 
				break;

			case 3:
				display(s); 
				break;

			default:
				printf("Enter valid choice\n");
		}

		printf("Continue? y or n\n");
		scanf(" %c", &ch);
	} while(ch=='y' || ch=='Y');
}